inspirehep/records/api/base.py

```diff
@@ -684,3 +720,173 @@ def _add_file(self, url, original_url=None, filename=None, **kwargs):
 
         metadata["url"] = file_path
         return metadata
+
+    def get_enhanced_data(self, serializer=None):
+        """Prepares serialized record for elasticsearch
+        Args:
+            serializer(Schema): Schema which should be used to serialize/enchance
+        Returns:
+            dict: Data serialized/enhanced by serializer.
+                Do no return data which is not specified in Schema
+
+        """
+        if not self.es_serializer and not serializer:
+            return None
+        if not serializer:
+            serializer_module = importlib.import_module(
+                self.__module__.replace("api", "marshmallow")
+            )
+            serializer = getattr(serializer_module, self.es_serializer)
+        return serializer().dump(self).data
+
+    def get_ui_data(self, serializer=None):
+        """Prepares serialized record for ui
+        Returns:
+            dict: Enhanced record data
+        """
+        if not self.ui_serializer and not serializer:
+            return None
+        if not serializer:
+            serializer_module = importlib.import_module(
+                self.__module__.replace("api", "marshmallow")
+            )
+            serializer = getattr(serializer_module, self.ui_serializer)
+        return serializer().dump(self).data
```

Comment:
isn't this exactly the same function as the previous one, except for the use of `self.ui_serializer` instead of of `self.es_serializer`?

Reply:
Yest it's very similar.